lib/app/app.dart

```diff
@@ -1,13 +1,24 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:todo_api/todo_api.dart';
 import 'package:todo_bloc/home/home.dart';
+import 'package:todo_repository/todo_repository.dart';
 
 class App extends StatelessWidget {
-  const App({Key? key}) : super(key: key);
+  final TodoApi todoApi;
+
+  const App({
+    Key? key,
+    required this.todoApi,
+  }) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    return const MaterialApp(
-      home: HomePage(),
+    return RepositoryProvider(
```

Comment:
Heads up! If we inject the repository in the constructor, then we can use `RepositoryProvider.value`